src/providers/virtualraster/qgsvirtualrasterprovider.cpp

```diff
@@ -336,6 +346,22 @@ int QgsVirtualRasterProvider::capabilities() const
   return capability;
 }
 
+QgsRasterIdentifyResult QgsVirtualRasterProvider::identify( const QgsPointXY &point, Qgis::RasterIdentifyFormat format, const QgsRectangle &boundingBox, int width, int height, int dpi )
+{
+  QgsRasterIdentifyResult result = QgsRasterDataProvider::identify( point, format, boundingBox, width, height, dpi );
+
+  // Checking the value returned by QgsRasterDataProvider::identify
+  // If it is NoData value, replace it with QVariant()
+  QMap<int, QVariant> bandValues = result.results();
```

Comment:
The QgsVirtualRasterProvider supports IdentifyValue only, and I think any other code that want to call QgsVirtualRasterProvider::identify should check the provider's capability before calling and that code should pass Qgis::RasterIdentifyFormat::Value as the format.
https://github.com/qgis/QGIS/blob/7d81fdd2dfe819629b461e9daa0bfb89be2ba61d/src/providers/virtualraster/qgsvirtualrasterprovider.cpp#L327-L331

When we look into the implementation of QgsRasterDataProvider::identify, the only case that `bandValues` will be empty is the format not equals to  Qgis::RasterIdentifyFormat::Value(line 276). However, that should not be happend acoording to the above explaination. Note that the band number of QgsVirtualRasterProvider is always 1, so the execution of QgsRasterDataProvider::identify ensures that `bandValues` has only one item. .https://github.com/qgis/QGIS/blob/7d81fdd2dfe819629b461e9daa0bfb89be2ba61d/src/core/raster/qgsrasterdataprovider.cpp#L269-L333

Since QgsVirtualRasterProvider has only one band, `mSrcNoDataValue`  is initialized with one value in the constructor, it can't be empty.

May be we should check if the `bandValues` is empty, and no need to check mSrcNoDataValue.

Reply:
This patch has committed.